src/node_api.cc

```diff
@@ -658,9 +658,16 @@ void napi_clear_last_error(napi_env env) {
 napi_status napi_set_last_error(napi_env env, napi_status error_code,
                                 uint32_t engine_error_code,
                                 void* engine_reserved) {
-  env->last_error.error_code = error_code;
-  env->last_error.engine_error_code = engine_error_code;
-  env->last_error.engine_reserved = engine_reserved;
+
+  if (env == nullptr) {
+    if (error_code == napi_ok) {
```

Comment:
As we've dropped `NAPI_PREAMBLE(env)` is some places, we've also dropped the preliminary `CHECK_ARG(env, env)` that was a part of it. I'm putting those back in.